addon/mixins/dom.js

```diff
@@ -86,9 +86,16 @@ function removeHandlerFromListenerData(handler) {
 export default Mixin.create({
   init() {
     this._super(...arguments);
+  },
+
+  get _listeners() {
```

Comment:
discussed briefly with @scalvert about this, we can't use real JS getters here for two reasons:

* it creates the getter on the POJO passed in to `Mixin.create()` which isn't what we want
* it makes it impossible to do the correct cleanup in `willDestroy` without creating the array during teardown

I proposed an alternative approach which would use a specific API for pushing things into the array:

```js
_trackListener(item) { 
  let listeners = this._listeners;
  if (!listeners) { 
    listeners = this._listeners = []; 
  }

  listeners.push(item);
}
```